binaries/opae.io/main.h

```diff
@@ -215,6 +216,33 @@ struct vfio_device {
 
     return b;
   }
+
+  #define VFIO_VF_TOKEN_LEN (16)
+  int set_vf_token(const char *vf_token)
+  {
+    struct vfio_device_feature *df;
+    int ret;
+
+    df = (struct vfio_device_feature *)
+      malloc(sizeof(struct vfio_device_feature) + VFIO_VF_TOKEN_LEN);
```

Comment:
Rather than malloc/free, could we use new/delete?

Reply:
I'm not sure new/delete can be used here. We are allocating memory for variable length data structure. In this case the 16 bytes of token data are appended to the rest of the basic ioctl data structure.

Reply:
I think you could do something like
```c++
 df = (struct vfio_device_feature *) new uint8_t[sizeof(struct vfio_device_feature) + VFIO_VF_TOKEN_LEN];
```